src/chains/ethereum/ethereum/tests/api/eth/createAccessList.test.ts

```diff
@@ -0,0 +1,418 @@
+import assert from "assert";
+import { EthereumProvider } from "../../../src/provider";
+import getProvider from "../../helpers/getProvider";
+import compile, { CompileOutput } from "../../helpers/compile";
+import { join } from "path";
+import { Quantity } from "@ganache/utils";
+import { Ethereum } from "../../../src/api-types";
+import { AccessList } from "@ganache/ethereum-transaction/src/access-lists";
+
+const encodeValue = (val: number | string) => {
+  return Quantity.from(val).toBuffer().toString("hex").padStart(64, "0");
+};
+
+async function deployContract(provider, from, code) {
+  await provider.send("eth_subscribe", ["newHeads"]);
+  const transactionHash = await provider.send("eth_sendTransaction", [
+    {
+      from,
+      data: code,
+      gas: "0xfffff"
+    } as any
+  ]);
+  await provider.once("message");
+
+  const receipt = await provider.send("eth_getTransactionReceipt", [
+    transactionHash
+  ]);
+
+  return receipt.contractAddress;
+}
+
+describe("api", () => {
+  describe("eth", () => {
+    describe("createAccessList", () => {
+      let contract: CompileOutput;
+      let provider: EthereumProvider;
+      let from, to, addr, encodedTo, encodedAddr: string;
+      let contractAddress: string;
+      let contractMethods: { [methodName: string]: string };
+      let slot, hexSlot: string;
+
+      before("compile", () => {
+        contract = compile(join(__dirname, "./contracts/Inspector.sol"), {
+          contractName: "Inspector"
+        });
+      });
+
+      before(async () => {
+        provider = await getProvider({});
+        [from, to, addr] = await provider.send("eth_accounts");
+
+        contractAddress = await deployContract(provider, from, contract.code);
+        contractMethods = contract.contract.evm.methodIdentifiers;
+        // make a random number between 0 and f to use as our slot for these tests
+        const random = Math.floor(Math.random() * 16).toString(16);
```

Comment:
You could use fast-check to generate this

Reply:
Whattttt? You don't love my super sophisticated mechanism for making robust tests??

Reply:
I was about to comment on the use of a random here. I don't really like it... but I don't think it's harmful here.